src/exchange.c

```diff
@@ -266,9 +260,14 @@ static int p11prov_ecdh_derive(void *ctx, unsigned char *secret,
 
     /* complete key template */
     if (ecdhctx->kdf_outlen) {
-        key_size = ecdhctx->kdf_outlen;
+        if (ecdhctx->kdf_outlen < outlen) {
+            key_size = ecdhctx->kdf_outlen;
+        }
```

Comment:
with this change, the `key_size` is kept on 0 in case of `ecdhctx->kdf_outlen >= outlen` instead of an error that was raised previously if I see right, which does not look correct.

Reply:
Uggh good catch.
I am not completely focused today, but I believe I should have fixed this now